src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/DateTimeEqualityRewriter.cs

```diff
@@ -65,7 +69,7 @@ public override Expression VisitMultiary(MultiaryExpression expression, object c
                 newExpressions.Add(expression.Expressions[^1]);
             }
 
-            return newExpressions == null ? expression : Expression.And(newExpressions);
+            return newExpressions == null ? expression : checkDateTimeAgainstPeriod ? Expression.Or(Expression.And(newExpressions), Expression.And(dateInPeriodExpression)) : Expression.And(newExpressions);
```

Comment:
nit: For clarity, I think we shouldn't nest ternary operators :)

Reply:
Simplified.